### DISCUSSION TITLE
child theme and github / netlify

### DISCUSSION THREAD

**Discussion by mroswell** [thuliteio/doks#346] [Q&A]
I had used someone else's docs theme, hosted on netlify. Then decided to switch over to doks. 

I want to push to the same repo I was using before. (At least I think that would be one way of not having to deal with DNS settings in netlify.) 

I don't know how .git/config should look. 

I don't quite understand upstream vs. I believe that some of the layouts only appear on your repository, and not locally. (at least I think that's the case.)

for now when I do:
`%  git push origin main`
I get:
```
To github.com:mroswell/hugonewbie.com.git
 ! [rejected]        main -> main (fetch first)
error: failed to push some refs to 'git@github.com:mroswell/hugonewbie.com.git'
hint: Updates were rejected because the remote contains work that you do
hint: not have locally. This is usually caused by another repository pushing
hint: to the same ref. You may want to first integrate the remote changes
hint: (e.g., 'git pull ...') before pushing again.
hint: See the 'Note about fast-forwards' in 'git push --help' for details.
```

My .git/config currently looks like this mess, after too many experiments. 
- How do things stay connected as a child theme? 
- How do I set .git/config? 
- How do I make sure it goes to `main` and not `master`?  
- And will the site still work on netlify? 

[core]
        repositoryformatversion = 0
        filemode = true
        bare = false
        logallrefupdates = true
        ignorecase = true
        precomposeunicode = true
[remote "origin"]
        url = git@github.com:mroswell/hugonewbie.com.git
        fetch = +refs/heads/*:refs/remotes/origin/*
        [branch "main"]
        remote = origin
        merge = refs/heads/main
#[branch "master"]
#       remote = origin
#       merge = refs/heads/master
#[remote "upstream"]
#       url = https://github.com/h-enk/doks-child-theme.git
#       fetch = +refs/heads/*:refs/remotes/upstream/*
#[branch "master"]
#       remote = upstream
#       merge = refs/heads/main
~

My troubles started when I went to push, and of course, I couldn't because it still had your repo as the remote. I quickly changed the remote, but somewhere along the way, I've lost my way...

**Comment by h-enk** [comment]
All is local (your project that is). Doks dependencies are listed _locally_ (in package-lock.json) + are installed (and thus available) _locally_ (including `@hyas/doks` which is `h-enk/doks` packaged as an npm module) .

I would start fresh (that would really be the easiest way):
1. `git clone https://github.com/h-enk/doks-child-theme.git my-doks-site`
2. `npm run init` (`rimraf .git && git init -b main`)
3. et cetera
